packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -0,0 +1,160 @@
+
+/**
+ * @class {@link LinkModel}
+ */
+export default class {
+  accountId: string;
+
+  sourceEntity: EntityModel;
+  linkTypeModel: LinkTypeModel;
+  targetEntity: EntityModel;
+
+  constructor({
+    accountId,
+    sourceEntity,
+    linkTypeModel,
+    targetEntity,
+  }: LinkModelConstructorParams) {
+    this.accountId = accountId;
+
+    this.sourceEntity = sourceEntity;
+    this.linkTypeModel = linkTypeModel;
+    this.targetEntity = targetEntity;
+  }
+
+  /**
+   * Create a link between a source and a target entity using a specific link
+   * type.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   * @param params.linkTypeModel the Link Type of the link
+   * @param params.targetEntity the target entity of the link
+   */
+  static async create(
+    graphApi: GraphApi,
+    {
+      accountId,
+      sourceEntity,
+      linkTypeModel,
+      targetEntity,
+    }: LinkModelCreateParams,
+  ): Promise<LinkModel> {
+    const {
+      data: { sourceEntityId, linkTypeUri, targetEntityId },
+    } = await graphApi.createLink(sourceEntity.entityId, {
+      accountId,
+      linkTypeUri: linkTypeModel.schema.$id,
+      targetEntityId: targetEntity.entityId,
+    });
+
+    const fetchedParams = {
+      sourceEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: sourceEntityId,
+      }),
+      linkTypeModel: await LinkTypeModel.get(graphApi, {
+        versionedUri: linkTypeUri,
+      }),
+      targetEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: targetEntityId,
+      }),
+    };
+
+    return new LinkModel({
+      accountId,
+      ...fetchedParams,
+    });
+  }
+
+  /**
+   * Get all outgoing links of a source entity.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   */
+  static async getAllOutgoing(
+    graphApi: GraphApi,
+    { sourceEntity }: { sourceEntity: EntityModel },
+  ): Promise<LinkModel[]> {
+    const { data: entityLinks } = await graphApi.getEntityLinks(
+      sourceEntity.entityId,
+    );
+    return Promise.all(
+      Object.entries(entityLinks.outgoing)
+        /**
+         * When target entity id is a list the link has many targets.
+         * This is not supported yet
+         * @todo support 1:N links
+         * */
+        .filter(([_, targetEntityId]) => typeof targetEntityId === "string")
+        .map(
+          async ([linkTypeUri, targetEntityId]) =>
+            new LinkModel({
+              accountId: sourceEntity.accountId,
+              linkTypeModel: await LinkTypeModel.get(graphApi, {
+                versionedUri: linkTypeUri,
+              }),
+              sourceEntity,
+              targetEntity: await EntityModel.getLatest(graphApi, {
+                /** @todo figure out what account ID we use here */
```

Comment:
When are we going to do these todos? I'm slightly concerned that we're throwing a lot of todos around the codebase without thinking too careful about their discoverability. We should probably start embedding their relevant asana links directly in the todo just so that when we pick up the asana task we can `ctrl+f` for all occurrences or something.

Reply:
Added link to task in ce907a39c4449b548374a9542a9533c9a93bdf37

Reply:
Thanks :)